tests/test.cpp

```diff
@@ -832,6 +832,44 @@ void UnicodeTest() {
                      "\\u5225\\u30B5\\u30A4\\u30C8\\x01\\x80\"}", true);
 }
 
+void UnicodeSurrogatesTest() {
+  flatbuffers::Parser parser;
+
+  TEST_EQ(
+    parser.Parse(
+      "table T { F:string; }"
+      "root_type T;"
+      "{ F:\"\\uD83D\\uDCA9\"}"), true);
+  auto root = flatbuffers::GetRoot<flatbuffers::Table>(
+    parser.builder_.GetBufferPointer());
+  auto string = root->GetPointer<flatbuffers::String *>(
+    sizeof(flatbuffers::soffset_t));
```

Comment:
`sizeof(flatbuffers::soffset_t)` is not the correct way to specify this. The offset is indeed 4, but that is actually because it is `sizeof(flatbuffers::voffset_t) * 2` (the first field after the 2 fixed vtable fields). You can make this code slightly cleaner by instead calling `FieldIndexToOffset(0)` to compute the offset, and for extra documentation declare the field as `F:string (id: 0);`

This isn't the cleanest